### PR TITLE
Block editor: rewrite selection clearer

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Currently the selection clearer works by catching focus on a wrapper. Because of this focusable element, it's trickier to remove the wrapper and share it with the click redirect wrapper. Additionally, it also tricky to remove in the context of an iframe where the `body` element cannot be truly focusable (because of how the browser handler focus loss, returning focus to the `body` element in that case, without firing any focus events).

The selection clearer can easily be rewritten to look for clicks instead of focus. There's already a precedent of doing so in the click redirect, where we redirect focus to the last block if the user clicks at the bottom of the editor. The clearer is basically the same thing, detecting clicks around the content to clear the block selection.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-selection-clearer/index.js [hunk 1]
```diff
@@ -19,20 +19,25 @@ export function useBlockSelectionClearer( ref ) {
 			return;
 		}
 
-		function onFocus() {
+		function onMouseDown( event ) {
+			// Only handle clicks on the canvas, not the content.
+			if ( event.target.closest( '.wp-block' ) ) {
```
Comment (ellatrix) [inline packages/block-editor/src/components/block-selection-clearer/index.js:24]: This condition could be removed eventually when the content is directly rendered into the wrapper the `mousedown` event is listened for. Currently there are a few wrappers in between the selection clearer and the block for which we also need to detect clicks.

Comment (youknowriad) [reply packages/block-editor/src/components/block-selection-clearer/index.js:24]: The  changes in this PR break the selection clearer in the widgets screen.  `.wp-block` must not be enough.

Ii also wonder why you're using mousedown, focusing outside (without  clicking) should also have   the same behavior right?

Comment (youknowriad) [reply packages/block-editor/src/components/block-selection-clearer/index.js:24]: I'm going to restore the previous  behavior for now to perform a patch release.

**Comment by youknowriad** [comment]
I think this kind of impactful PRs could use more reviews.

File: packages/edit-post/src/components/visual-editor/index.js [hunk 2]
```diff
@@ -64,7 +64,6 @@ export default function VisualEditor() {
 			<div
 				ref={ ref }
 				className="editor-styles-wrapper"
-				tabIndex="-1"
```
Comment (youknowriad) [inline packages/edit-post/src/components/visual-editor/index.js:67]: Do we know why this was here and why this was removed?

**Pull request merged**